packages/react-jss/src/withStyles.js

```diff
@@ -79,6 +77,61 @@ export default function withStyles<Theme: {}, S: Styles<Theme>>(
       // $FlowFixMe
       static defaultProps = {...InnerComponent.defaultProps}
 
+      static getSheetClasses(sheet, dynamicRules: ?DynamicRules) {
+        const classes = {}
+
+        // $FlowFixMe
+        for (const key in sheet.styles) {
+          classes[key] = sheet.classes[key]
+
+          if (dynamicRules && key in dynamicRules) {
+            classes[key] += ` ${sheet.classes[dynamicRules[key].key]}`
+          }
+        }
+
+        return classes
+      }
+
+      static updateDynamicRules(props: HOCProps<Theme, Props>, {dynamicRules, sheet}: State) {
+        if (!sheet) {
+          return
+        }
+
+        for (const key in dynamicRules) {
```

Comment:
not sure why you need this loop? if you don't pass the first argument as a key, JSS will already do the same it seems

Reply:
Ah got it, you need to only update rules from a specific component instance, since that sheet will contain all dynamic values for all instances. We should add a comment, this isn't obvious.

Reply:
Will add a comment